hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/DefaultTraceEntityAccessor.java

```diff
@@ -71,50 +70,44 @@ private boolean isExcludedEntityType(final EntityType entityType) {
   }
 
   private void writeEntityIfExists(EntityType entityType, StructuredTrace trace, Event span) {
-    this.buildEntity(entityType, trace, span)
-        .subscribe(
-            entity -> {
-              UpsertCondition upsertCondition =
-                  this.buildUpsertCondition(entityType, trace, span)
-                      .defaultIfEmpty(UpsertCondition.getDefaultInstance())
-                      .blockingGet();
-
-              this.entityDataClient.createOrUpdateEntityEventually(
-                  RequestContext.forTenantId(this.traceAttributeReader.getTenantId(span)),
-                  entity,
-                  upsertCondition,
-                  this.writeThrottleDuration);
-            });
+    Optional<Entity> entityOptional = this.buildEntity(entityType, trace, span);
+    if (entityOptional.isEmpty()) {
+      return;
+    }
+    // TODO: in follow up PR batch eventual entity writes for single tenant
+    this.entityDataClient.createOrUpdateEntityEventually(
+        RequestContext.forTenantId(this.traceAttributeReader.getTenantId(span)),
+        entityOptional.get(),
+        this.buildUpsertCondition(entityType, trace, span)
+            .orElse(UpsertCondition.getDefaultInstance()),
```

Comment:
@aaron-steinfeld I'm not sure if we need an upsert call with default instance. 

I initially used if present only then call updateEntityEventually, couple of tests were verifying the call to `createOrUpdateEntityEventually` happens irrespective of presence of upsert condition. I think the timestamp early exit condition in build of upsert is being hit.

If no particular reason to include this default instance upsert calls, I'm inclined to remove it and use if present instead of orElse

Reply:
Do we need it to handle the create entity case when an entity appears for the first time?

Reply:
> I initially used if present only then call updateEntityEventually, couple of tests were verifying the call to createOrUpdateEntityEventually happens irrespective of presence of upsert condition.
I believe the timestamp early exit condition in build of upsert is being hit in those tests and causing no invoke when using ifPresent.
> 
> If no particular reason to include this default instance upsert calls, I'm inclined to remove it and use if present instead of orElse

Not positive I'm following here. The lack of an upsert condition doesn't tell us the entity shouldn't be upserted does it? I thought it was dependent on the entity type's config. Some entities may have unconditional upsert.

> Do we need it to handle the create entity case when an entity appears for the first time?

Yep, but upsert should do that already.

Reply:
> Not positive I'm following here. The lack of an upsert condition doesn't tell us the entity shouldn't be upserted does it? I thought it was dependent on the entity type's config. Some entities may have unconditional upsert.

Don't have context on this. @skjindal93 ?